pubsub/subscription.go

```diff
@@ -786,14 +936,21 @@ func (s *Subscription) Config(ctx context.Context) (SubscriptionConfig, error) {
 
 // SubscriptionConfigToUpdate describes how to update a subscription.
 type SubscriptionConfigToUpdate struct {
-	// If non-nil, the push config is changed. Cannot be set at the same time as BigQueryConfig.
+	// If non-nil, the push config is changed. At most one of PushConfig, BigQueryConfig, or CloudStorageConfig
+	// can be set.
 	// If currently in push mode, set this value to the zero value to revert to a Pull based subscription.
 	PushConfig *PushConfig
 
-	// If non-nil, the bigquery config is changed. Cannot be set at the same time as PushConfig.
+	// If non-nil, the bigquery config is changed. At most one of PushConfig, BigQueryConfig, or CloudStorageConfig
+	// can be set.
 	// If currently in bigquery mode, set this value to the zero value to revert to a Pull based subscription,
 	BigQueryConfig *BigQueryConfig
 
+	// If non-nil, the Cloud Storage config is changed. At most one of PushConfig, BigQueryConfig, or CloudStorageConfig
+	// can be set.
+	// If currently in CloudStorage mode, set this value to the zero value to revert to a Pull based subscription,
```

Comment:
Optional: "the zero value" sounds a bit odd, does it make sense to say "nil" instead? I realize we already have this wording in place for the PushConfig comment so feel free to push back on this.

Reply:
Zero value is correct here since nil means we're not trying to update he config. Setting a zero value (aka empty) CloudStorageConfig is how we're able to clear it. We translate this to a `nil` CloudStorageConfig before passing it into the update request.